stock_picking_auto_create_lot_qty/tests/test_stock_picking_auto_create_lot_qty.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (C) 2023 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo.exceptions import UserError
+from odoo.tests import SavepointCase
+
+from .common import CommonStockPickingAutoCreateLotQty
+
+
+class TestStockPickingAutoCreateLotQty(
+    CommonStockPickingAutoCreateLotQty, SavepointCase
+):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # Create a new product and set the following properties:
+        # * Create lot every 5 units
+        # * Only multiples allowed = True
+        cls.product_serial_auto_qty_5_multiples_allowed_true = cls._create_product(
+            tracking="lot", auto=True, every_n=5, multiple_allow=True
+        )
+        # * Only multiples allowed = False
+        cls.product_serial_auto_qty_5_multiples_allowed_false = cls._create_product(
+            tracking="lot", auto=True, every_n=5, multiple_allow=False
+        )
+        cls.picking_type_in.auto_create_lot = True
+
+        # * tracking "lot" to "none"
+        cls.product_no_tracking = cls._create_product(
+            tracking="lot", auto=True, every_n=5, multiple_allow=False
+        )
+        # * tracking every_n = 0
+        cls.product_0_every_n = cls._create_product(
+            tracking="lot", auto=True, every_n=0, multiple_allow=False
+        )
+
+    def test_multiples_allowed_false(self):
+        self._create_picking()
+        self._create_move(
+            product=self.product_serial_auto_qty_5_multiples_allowed_false, qty=8.0
+        )
+        self.picking.button_validate()
+        # Check the display field
+        move = self.picking.move_lines.filtered(
+            lambda m: m.product_id
+            == self.product_serial_auto_qty_5_multiples_allowed_false
+        )
+        self.assertFalse(move.display_assign_serial)
+
+        # Search for serials
+        lot = self.env["stock.production.lot"].search(
+            [
+                (
+                    "product_id",
+                    "=",
+                    self.product_serial_auto_qty_5_multiples_allowed_false.id,
+                )
+            ]
+        )
+        self.assertEqual(len(lot), 2)
+
+    def test_multiples_allowed_true(self):
+        self._create_picking()
+        self._create_move(
+            product=self.product_serial_auto_qty_5_multiples_allowed_true, qty=8.0
+        )
+        # check for error
+        self.assertRaises(UserError, self.picking.button_validate)
```

Comment:
Please add to this test a move with qty 15 and check that it works correctly

Reply:
Thanks for the feedback, I will of course add a test so that all cases are covered by tests.